src/backend/vulkanll/src/command.rs

```diff
@@ -60,16 +60,50 @@ impl CommandBuffer {
         unimplemented!()
     }
 
-    fn copy_buffer(&mut self, src: &native::Buffer, dst: &native::Buffer, regions: Option<&[command::BufferCopy]>) {
+    fn copy_buffer(&mut self, src: &native::Buffer, dst: &native::Buffer, regions:&[command::BufferCopy]) {
         unimplemented!()
     }
 
-    fn copy_image(&mut self, src: &native::Image, dest: &native::Image) {
+    fn copy_image(&mut self, src: &native::Image, dst: &native::Image) {
         unimplemented!()
     }
 
-    fn copy_buffer_to_image(&mut self) {
-        unimplemented!()
+    fn copy_buffer_to_image(&mut self, src: &native::Buffer, dst: &native::Image, layout: memory::ImageLayout, regions: &[command::BufferImageCopy]) {
+        let regions = regions.iter().map(|region| {
+            let subresource_layers = vk::ImageSubresourceLayers {
+                aspect_mask: vk::IMAGE_ASPECT_COLOR_BIT, // TODO
+                mip_level: region.image_mip_level as u32,
+                base_array_layer: region.image_base_layer as u32,
+                layer_count: region.image_layers as u32,
+            };
+
+            vk::BufferImageCopy {
+                buffer_offset: region.buffer_offset,
+                buffer_row_length: 0,
+                buffer_image_height: 0,
+                image_subresource: subresource_layers,
+                image_offset: vk::Offset3D {
+                    x: region.image_offset.x,
+                    y: region.image_offset.y,
+                    z: region.image_offset.z,
+                },
+                image_extent: vk::Extent3D {
+                    width:  region.image_extent.width,
+                    height: region.image_extent.height,
+                    depth:  region.image_extent.depth,
+                },
+            }
+        }).collect::<Vec<_>>();
```

Comment:
we'd like to avoid heap allocations in the future, but this is totally fine at this stage of corell